From ac3ff19483556fea4ec5ce4c486493f24de375f2 Mon Sep 17 00:00:00 2001
From: Peter Colberg <peter.colberg@intel.com>
Date: Wed, 31 Jan 2024 18:27:28 -0500
Subject: [PATCH] fpga: dfl-cxl-cache: depend on DRM

This resolves an undefined reference to drm_clflush_virt_range()
when building a kernel without modules and minimal configuration.

Signed-off-by: Peter Colberg <peter.colberg@intel.com>
---
 configs/dfl-config   | 1 +
 drivers/fpga/Kconfig | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/configs/dfl-config b/configs/dfl-config
index a7a5e93fb918..09565b1f4f2b 100644
--- a/configs/dfl-config
+++ b/configs/dfl-config
@@ -61,6 +61,7 @@ CONFIG_PTP_DFL_TOD=m
 
 # CXL cache support
 
+CONFIG_DRM=m
 CONFIG_FPGA_DFL_CXL_CACHE=m
 
 # Test configs - not required for production environments
diff --git a/drivers/fpga/Kconfig b/drivers/fpga/Kconfig
index 7031cb5485fb..a35eb285fde3 100644
--- a/drivers/fpga/Kconfig
+++ b/drivers/fpga/Kconfig
@@ -296,7 +296,7 @@ config FPGA_M10_BMC_SEC_UPDATE
 
 config FPGA_DFL_CXL_CACHE
 	tristate "Intel CXL cache driver"
-	depends on FPGA_DFL
+	depends on DRM && FPGA_DFL
 	help
 	  This is the driver for CXL cache Accelerated Function Unit
 	  (AFU) which provides interfaces to MMIO region and dma buffers.
